pkg/kv/kvprober/kvprober.go

```diff
@@ -428,7 +429,7 @@ func (p *Prober) readProbeImpl(ctx context.Context, ops proberOpsI, txns proberT
 	if tracingEnabled.Get(&p.settings.SV) {
 		leaseholder := p.returnLeaseholderInfo(finishAndGetRecording())
 		ctx = logtags.AddTag(ctx, "leaseholder", leaseholder)
-		log.Health.Infof(ctx, "kv.Get(%s), r=%v having likely leaseholder=%s returned success in %v", step.Key, step.RangeID, leaseholder, d)
+		log.Health.Infof(ctx, "kv.Get(%s), r=%v having likely leaseholder=%s returned success in %v", step.Key, step.RangeID, redact.Safe(leaseholder), d)
```

Comment:
This is not the correct way to implement safe redaction in this scenario. What you want to do is make `returnLeaseholderInfo` return a redactable object. In this case it's returning a string which previously contained redaction data that was stripped away here: https://github.com/cockroachdb/cockroach/blob/2444856f86e45c16626ac535068b5a1405476325/pkg/kv/kvprober/kvprober.go#L570

This is an unsafe operation and should be modified. What you likely want is for `findLastLogMessage` to return a `RedactableString`. When you pass that string as an arg into `Infof` it will get redacted correctly.

Reply:
TFTR! I made the change to return a `RedactableString`

Reply:
@dhartunian I don't understand how this will unredact the nodeId. IIUC, if we don't strip the markers and mark it redact safe, ddog will still show it as redacted, right?

Reply:
@VishalJaishankar You are right. `RedactableString` is an extended string type that is meant to hold a mix of redacted and un-redacted parts. So, just casting the nodeID to this type will not unredact it (because it's always unsafe by default). See [this comment ](https://github.com/cockroachdb/redact/blob/b7e45562c4d0d78dd4d6c7f08f65e8220a0d7b77/internal/markers/markers.go#L24-L26). `RedactableString` are constructed using functions like `redact.Sprint` and `redact.Sprintf`.  This lets users decide which part of the string is safe and which part is unsafe.

Looking at the [doc strings of `redact.Sprintf`](https://github.com/cockroachdb/redact/blob/b7e45562c4d0d78dd4d6c7f08f65e8220a0d7b77/markers_print.go#L33-L39) I would just return `redact.Sprintf(leaseholder[1])`. The 1st arg (`format`) is always considered safe. `redact.Sprint(redact.Safe(leaseholder[1]))` is another way. But not recommended (it's a more expensive operation).

Reply:
Thanks for the detailed context @arjunmahishi , made the changes, could you PTAL